caom2utils/caom2utils/fits2caom2.py

```diff
@@ -835,112 +835,104 @@ def configure_time_axis(self, axis, override=True):
-            if time_axis in ['1', '2'] or energy_axis in ['1', '2'] or \
-                    polarization_axis in ['1', '2'] or obs_axis in ['1', '2']:
+            if axis_info['time'][0] in ['1', '2'] or \
+                    axis_info['energy'][0] in ['1', '2'] or \
+                    axis_info['polarization'][0] in ['1', '2'] or \
+                    axis_info['obs'][0] in ['1', '2']:
                 raise ValueError('Cannot determine the positional axis')
             else:
                 self.configure_position_axes(('1', '2'), False)
 
-        if time_axis_expected:
-            self.configure_time_axis(time_axis, False)
-
-        if energy_axis_expected:
-            self.configure_energy_axis(energy_axis, False)
-
-        if polarization_axis_expected:
-            self.configure_polarization_axis(polarization_axis, False)
+        if axis_info['time'][1]:
+            self.configure_time_axis(axis_info['time'][0], False)
+        if axis_info['energy'][1]:
+            self.configure_energy_axis(axis_info['energy'][0], False)
+        if axis_info['polarization'][1]:
+            self.configure_polarization_axis(axis_info['polarization'][0],
+                                             False)
+        if axis_info['obs'][1]:
+            self.configure_observable_axis(axis_info['obs'][0], False)
+
+    def _guess_axis_info_from_ctypes(self, lookup, counter, axis_info):
+        if lookup.startswith('Chunk.energy'):
+            axis_info['energy'] = (counter, True)
+            self._check_dual_sets(axis_info, 'energy')
+        elif lookup.startswith('Chunk.polarization'):
+            axis_info['polarization'] = (counter, True)
+            self._check_dual_sets(axis_info, 'polarization')
+        elif lookup.startswith('Chunk.time'):
+            axis_info['time'] = (counter, True)
+            self._check_dual_sets(axis_info, 'time')
+        elif lookup.startswith('Chunk.position') and lookup.endswith(
+                'axis1.ctype'):
+            axis_info['ra'] = (counter, True)
+            self._check_dual_sets(axis_info, 'ra')
+        elif lookup.startswith('Chunk.position') and lookup.endswith(
+                'axis2.ctype'):
+            axis_info['dec'] = (counter, True)
+            self._check_dual_sets(axis_info, 'dec')
+        elif lookup.startswith('Chunk.observable'):
+            axis_info['obs'] = (counter, True)
+            self._check_dual_sets(axis_info, 'obs')
+        else:
+            raise ValueError(
+                'Unrecognized axis type: {}'.format(lookup))
 
-        if obs_axis_expected:
-            self.configure_observable_axis(obs_axis, False)
+    def _check_dual_sets(self, axis_info, lookup):
+        for axis in axis_info:
+            if axis == axis_info[lookup]:
+                continue
+            if axis[1] and axis_info[lookup][0] == axis[0]:
+                raise ValueError('Conflicting indices for {} and {}'.format(
+                    axis, lookup))
```

Comment:
Wondering if this could be avoided if the axis_info is filled in with the available info in the blueprint followed by adding missing default info.

Reply:
See the reworked implementation of the method _guess_axis_info_from_plan